clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java

```diff
@@ -15,57 +15,167 @@
+                  TaskDefinition taskDef = new TaskDefinition().withTaskDefinitionArn(testArn);
+                  return new RegisterTaskDefinitionResult().withTaskDefinition(taskDef);
+                });
+    when(mockECS.createService(any(CreateServiceRequest.class)))
+        .thenReturn(
+            new CreateServiceResult().withService(new Service().withServiceName("createdService")));
+
+    when(mockAwsProvider.getAmazonEcs(
+            any(NetflixAmazonCredentials.class), anyString(), anyBoolean()))
+        .thenReturn(mockECS);
+
+    // mock ELB responses
+    when(mockELB.describeTargetGroups(any(DescribeTargetGroupsRequest.class)))
+        .thenAnswer(
+            (Answer<DescribeTargetGroupsResult>)
+                invocation -> {
+                  DescribeTargetGroupsRequest request =
+                      (DescribeTargetGroupsRequest) invocation.getArguments()[0];
+                  String testArn =
+                      "arn:aws:elasticloadbalancing:::targetgroup/"
+                          + request.getNames().get(0)
+                          + "/76tgredfc";
+                  TargetGroup testTg = new TargetGroup().withTargetGroupArn(testArn);
+
+                  return new DescribeTargetGroupsResult().withTargetGroups(testTg);
+                });
+
+    when(mockAwsProvider.getAmazonElasticLoadBalancingV2(
+            any(NetflixAmazonCredentials.class), anyString(), anyBoolean()))
+        .thenReturn(mockELB);
+  }
+
   @DisplayName(
       ".\n===\n"
           + "Given description w/ inputs, EC2 launch type, and legacy target group fields, "
           + "successfully submit createServerGroup operation"
           + "\n===")
   @Test
-  public void createServerGroupOperationTest() throws IOException {
+  public void createServerGroup_InputsEc2LegacyTargetGroupTest()
+      throws IOException, InterruptedException {
 
     // given
     String url = getTestUrl(CREATE_SG_TEST_PATH);
     String requestBody = generateStringFromTestFile("/createServerGroup-inputs-ec2.json");
+    String expectedServerGroupName = "ecs-integInputsEc2LegacyTargetGroup";
     setEcsAccountCreds();
 
     // when
-    given()
-        .contentType(ContentType.JSON)
-        .body(requestBody)
-        .when()
-        .post(url)
-        // then
-        .then()
-        .statusCode(200)
-        .contentType(ContentType.JSON)
-        .body("id", notNullValue())
-        .body("resourceUri", containsString("/task/"));
+    String taskId =
+        given()
+            .contentType(ContentType.JSON)
+            .body(requestBody)
+            .when()
+            .post(url)
+            .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
+            .body("id", notNullValue())
+            .body("resourceUri", containsString("/task/"))
+            .extract()
+            .path("id");
+
+    retryUntilTrue(
```

Comment:
Do we need retry logic in tests, you wouldn't be making calls outside the scope of clouddriver, so should any call take too long?

Reply:
for context: these tests run against a live/running instance of clouddriver, which begins execution of the `createServerGroupAtomicOperation` workflow _after_ successful submission of the task. 

Asserting against the mocks directly after submitting the operation led to errors b/c often the mocked calls had not yet been invoked. It's still pretty fast, so adding a brief `Thread.sleep()` before verifying could potentially solve this as well (it did during testing), but by retrying on the provided task ID we're more closely mocking how `Orca` would poll for task completion  + implicitly testing that the provided ID matches the intended operation.

Reply:
Makes sense, I'm just hesitant in adding additional retry logic that's tightly coupled to another service (orca in this case) as that can change.

Reply:
well to be clear, we're not mocking anything Orca-specific here, it's just that the operation is asynchronous so anything submitting one (aka Orca) would have to query the task ID later to figure out if it succeeded or not.

Reply:
Since the retry count is 3, I'm okay with this (I'm hesitant in hardcoding retry logic that could potentially get the tests into a weird state).